### PR TITLE
Likes component

### PR DESCRIPTION
**Pull request by RussellSk**

### CODE REVIEW

File: core/src/main/java/com/exadel/core/services/impl/LikesServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        try (Connection connection = databaseService.getConnection(DATA_SOURCE)) {
+            final PreparedStatement statement = connection.prepareStatement(GET_USER_PRODUCT_LIKES);
+            statement.setString(1, product);
+            statement.setString(2, user);
+            try (ResultSet result = statement.executeQuery()) {
+                // if user already give a like/dislike
+                if (result.next()) {
+                    // convert like to dislike or dislike to like
+                    boolean currentLikeType = result.getBoolean(4);
+                    if (likeType != currentLikeType) {
```
Comment (AndriiHolovianko) [inline core/src/main/java/com/exadel/core/services/impl/LikesServiceImpl.java:84]: Maybe it's better to extract methods in order to simplify method

File: core/src/main/java/com/exadel/core/services/impl/LikesServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                    // else insert new like/dislike
+                    final PreparedStatement insertStatement = connection.prepareStatement(INSERT_NEW_LIKE);
+                    insertStatement.setString(1, product);
+                    insertStatement.setString(2, user);
+                    insertStatement.setBoolean(3, likeType);
+                    insertStatement.execute();
+                }
+            }
+        } catch (SQLException | DataSourceNotFoundException exception) {
+            log.error("handleLike: {}", getClass().getName(), exception);
```
Comment (AndriiHolovianko) [inline core/src/main/java/com/exadel/core/services/impl/LikesServiceImpl.java:103]: Please write a detailed message. e.g. Exception during 'like' handling operation

File: core/src/main/java/com/exadel/core/servlets/LikesServlet.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    @Override
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        try {
+            // Get request parameters
+            String product = request.getParameter("product");
+            boolean likeType = Boolean.parseBoolean(request.getParameter("like_type"));
+            String userIpAddress = getUserIp(request);
+
+            // Process like/dislike persistence
```
Comment (AndriiHolovianko) [inline core/src/main/java/com/exadel/core/servlets/LikesServlet.java:42]: It's better to move such logic into service

File: core/src/main/java/com/exadel/core/servlets/LikesServlet.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            jsonObject.put("status", "success");
+            jsonObject.put("likesCount", likesCount);
+            jsonObject.put("dislikesCount", dislikesCount);
+
+            // Write prepared JSON to response
+            response.setContentType("application/json");
+            response.getWriter().write(jsonObject.toString());
+
+        } catch (Exception exception) {
+            log.error("doPost: {}", getServletName(), exception);
```
Comment (AndriiHolovianko) [inline core/src/main/java/com/exadel/core/servlets/LikesServlet.java:64]: It's better to not write an exception to user response(it's not save)

File: core/src/main/java/com/exadel/core/servlets/LikesServlet.java (hunk 3 above)
Comment (RussellSk) [reply core/src/main/java/com/exadel/core/servlets/LikesServlet.java:42]: Should I move getUserIp method into different service?

**Pull request merged**